parsers/ES.py

```diff
@@ -215,35 +309,36 @@ def fetch_exchange(
 ) -> list[dict]:
     sorted_zone_keys = ZoneKey("->".join(sorted([zone_key1, zone_key2])))
     check_valid_parameters(sorted_zone_keys, session, target_datetime)
-
-    if isinstance(target_datetime, datetime):
-        date = target_datetime.strftime("%Y-%m-%d")
-    else:
-        date = target_datetime
-
     ses = session or Session()
 
-    responses: list[Response] = EXCHANGE_FUNCTION_MAP[sorted_zone_keys](ses).get_all(
-        date
+    responses = fetch_island_data(
+        EXCHANGE_FUNCTION_MAP[sorted_zone_keys],
+        ses,
+        target_datetime,
     )
 
     exchangeList = ExchangeList(logger)
-    for response in responses:
+    for event in responses:
+        exchanges = {}
+        for key in event:
+            if key in EXCHANGE_PARSE_MAPPING.keys():
+                exchanges[EXCHANGE_PARSE_MAPPING[key]] = event[key]
+            elif key not in PRODUCTION_PARSE_MAPPING and key != "dem" and key != "hid":
```

Comment:
```suggestion
            elif key not in PRODUCTION_PARSE_MAPPING and key != "dem" and key != "hid" and key != "datetime":
```

I would suggest declaring a list for this somehwere as it is reused in several places, or move all the warnings into the preprocess step by checking all the keys there (dem, production and exchange).